src/FondOfSpryker/Zed/CompanyUserCompanyAssigner/Business/Model/CompanyRoleInterface.php

```diff
@@ -0,0 +1,26 @@
+<?php
+
+namespace FondOfSpryker\Zed\CompanyUserCompanyAssigner\Business\Model;
+
+use Generated\Shared\Transfer\CompanyUserTransfer;
+
+interface CompanyRoleInterface
+{
+    /**
+     * @param \Generated\Shared\Transfer\CompanyUserTransfer $companyUserTransfer
+     *
+     * @return void
+     */
+    public function updateNonManufacturerUsersCompanyRole(
+        CompanyUserTransfer $companyUserTransfer
+    ): void;
+
+    /**
+     * @param \Generated\Shared\Transfer\CompanyUserTransfer $companyUserTransfer
+     *
+     * @return array<int, array<string, mixed>>
+     */
+    public function findCompanyUsersWithDiffCompanyRolesAsManufacturer(
```

Comment:
Use an own interface (like CompanyUserReaderInterface) for this method.

Reply:
@daniel-rose because i renamed the Model/CompanRole to Manager/CompanyRoleManager as suggested, i do not know it is still neccessary to create a separat Class for the method. I just checked how spryker does. But of course we can discuss it.